docs/design/services/dynamodb/high-level-library/prototype/option-1/sync/Prototype.java

```diff
@@ -0,0 +1,1025 @@
+/**
+ * The entry-point for all DynamoDB client creation. All Java Dynamo features will be accessible through this single class. This
+ * enables easy access to the different abstractions that the AWS SDK for Java provides (in exchange for a bigger JAR size).
+ *
+ * <p>
+ *     <b>Maven Module Location</b>
+ *     This would be in a separate maven module (software.amazon.awssdk:dynamodb-all) that depends on all other DynamoDB modules
+ *     (software.amazon.awssdk:dynamodb, software.amazon.awssdk:dynamodb-document). Customers that only want one specific client
+ *     could instead depend directly on the module that contains it.
+ * </p>
+ */
+@ThreadSafe
+public interface DynamoDb {
+    /**
+     * Create a low-level DynamoDB client with default configuration. Equivalent to DynamoDbClient.create().
+     * Already GA in module software.amazon.awssdk:dynamodb.
+     */
+    DynamoDbClient client();
+
+    /**
+     * Create a low-level DynamoDB client builder. Equivalent to DynamoDbClient.builder().
+     * Already GA in module software.amazon.awssdk:dynamodb.
+     */
+    DynamoDbClientBuilder clientBuilder();
+
+    /**
+     * Create a high-level "document" DynamoDB client with default configuration.
+     *
+     * Usage Example:
+     * <code>
+     *     try (DynamoDbDocumentClient client = DynamoDb.documentClient()) {
```

Comment:
nit: not related, but something to think about, we might not want to use try-with-resources in the code snippet  in the Javadoc because that might encourage people to create new client per request

Reply:
Other hand: not using try-with-resources in code snippets might encourage people not to close their clients.

Point taken, though.